homeassistant/components/nut/config_flow.py

```diff
@@ -241,16 +242,26 @@ async def async_step_init(self, user_input=None):
             CONF_SCAN_INTERVAL, DEFAULT_SCAN_INTERVAL
         )
 
-        info = await validate_input(self.hass, self.config_entry.data)
+        errors = {}
+        info = {}
+        try:
+            info = await validate_input(self.hass, self.config_entry.data)
+        except CannotConnect:
+            errors[CONF_BASE] = "cannot_connect"
```

Comment:
Should we abort if we cannot connect since we can't validate the resource list will be correct?

Reply:
Mhhh 🤔 my thoughts were too muddled for such a smart solution to occur to me 🙈 
I will change it tomorrow (CET)

Reply:
On the other hand, with that the user is still able to configure the scan interval and the list of resources is limited to the current active one ... so they should be valid 🤔

Reply:
I still think it should abort as there is no point in leaving them in an options flow that can't be validated.

Reply:
a simple `return False` causes tracebacks, therefore I decided to return an form without a schema, but `errors` set to inform the user, that NUT server is currently not available. When clicking on Submit, no configuration changes were made.
![grafik](https://user-images.githubusercontent.com/35783820/109346180-99172300-7871-11eb-8232-23807cb840cf.png)

Overall, thee is no clean way (_or I do not know it_) to abort an options flow :thinking:

Reply:
Here is how an abort happens in `homekit` options:

```
    async def async_step_yaml(self, user_input=None):
        """No options for yaml managed entries."""
        if user_input is not None:
            # Apparently not possible to abort an options flow
            # at the moment
            return self.async_create_entry(title="", data=self.config_entry.options)

        return self.async_show_form(step_id="yaml")
```

Reply:
Mhhh 🤔 ok, I will give it a try and report back ...

Reply:
I have done a combination of both solutions ... result:

1. UPS is reachable
![grafik](https://user-images.githubusercontent.com/35783820/110034113-615d1f00-7d3a-11eb-8360-b7103b89506d.png)

2. UPS is not reachable
![grafik](https://user-images.githubusercontent.com/35783820/110034178-746fef00-7d3a-11eb-9dba-a8c4873844c6.png)